app/view_objects/results_view.rb

```diff
@@ -201,19 +201,43 @@ def suggested_search_visible?
   end
 
   def suggested_search_links
-    @suggested_search_links ||= begin
-                                  all_links = []
-                                  radii_for_suggestions.each do |radius|
-                                    break if filter_links(all_links).count == 2
-
-                                    all_links << SuggestedSearchLink.new(
-                                      radius: radius,
-                                      count: course_counter(radius_to_check: radius),
-                                      parameters: query_parameters_with_defaults,
-                                    )
+    if with_salaries?
+      all_links = []
+
+      first_link = suggested_search_link_including_unsalaried(current_radius: radius)
+
+      all_links << first_link if first_link.present?
+
+      radii_for_suggestions.each do |radius|
+        break if all_links.count == 2
+
+        count = course_counter(radius_to_check: radius)
+
+        if count > course_count
+          all_links << SuggestedSearchLink.new(
+            radius: radius,
+            count: count,
+            parameters: query_parameters_with_defaults,
+          )
+        end
+      end
+
+      @suggested_search_links ||= all_links
+    else
+      @suggested_search_links ||= begin
```

Comment:
Is there a need for the entire block to be or-assigned? Can't you just run that code before and or-assign the `filter_links(all_links)` against it?